sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxReportSensor.java

```diff
@@ -142,25 +147,36 @@ protected String getStringProperty(String name, String def) {
     return value;
   }
 
-  public static List<File> getReports(Settings settings,
-    File moduleBaseDir,
-    String reportPathPropertyKey) {
+  public static List<File> getReports(Settings settings, final File moduleBaseDir,
+      String reportPathPropertyKey) {
 
-    String[] reportPaths = settings.getStringArray(reportPathPropertyKey);
     List<File> reports = new ArrayList<>();
-    if (reportPaths.length > 0) {
-      for (String reportPath : reportPaths) {
-        reportPath = FilenameUtils.normalize(reportPath);
-        File singleFile = new File(reportPath);
-        if (singleFile.exists()) {
-          reports.add(singleFile);
-        } else {
-          CxxUtils.LOG.debug("Using pattern '{}' to find reports", reportPath);
-          CxxUtils.GetReportForBaseDirAndPattern(moduleBaseDir.getPath(), reportPath, reports);
-          if (reports.isEmpty()) {
-            CxxUtils.LOG.warn("Cannot find a report for '{}={}'", reportPathPropertyKey, reportPath);
+
+    List<String> includes = Arrays.asList(settings.getStringArray(reportPathPropertyKey));
+    if (!includes.isEmpty()) {
+      includes = Lists.transform(includes, new Function<String, String>() {
+        @Override
+        public String apply(@Nullable String path) {
+          if (path == null){
+            return null;
+          }
+          path = FilenameUtils.normalize(path);
+          if (new File(path).isAbsolute()) {
+            return path;
           }
+          return FilenameUtils.normalize(moduleBaseDir.getAbsolutePath() + File.separator + path);
         }
+      });
+
+      CxxUtils.LOG.debug("Normalized report includes: {}", includes);
+
+      DirectoryScanner directoryScanner = new DirectoryScanner();
+      directoryScanner.setIncludes(includes.toArray(new String[includes.size()]));
+      directoryScanner.scan();
+
+      for (String found : directoryScanner.getIncludedFiles()) {
+        CxxUtils.LOG.debug("Adding report {}", found);
+        reports.add(new File(found));
       }
```

Comment:
why not adding the warning again when `reports` is empty after for loop?

e.g.
```Java
CxxUtils.LOG.warn("Cannot find a report for '{}'", reportPathPropertyKey);
```